src/MongoDB.Driver.Core/Core/Configuration/ConnectionSettings.cs

```diff
@@ -27,14 +27,9 @@ namespace MongoDB.Driver.Core.Configuration
     /// </summary>
     public class ConnectionSettings
     {
-        #region static
-        // static fields
-        private static readonly IReadOnlyList<IAuthenticator> __noAuthenticators = new IAuthenticator[0];
-        #endregion
-
         // fields
         private readonly string _applicationName;
-        private readonly IReadOnlyList<IAuthenticator> _authenticators;
+        private readonly IAuthenticatorsFactory _authenticatorsFactory;
```

Comment:
Replace with:

    private readonly IReadOnlyList<IAuthenticatorFactory> _authenticatorFactories;

and make all other corresponding changes to the class.